### ISSUE TITLE
Need negative semantics for K

### ISSUE THREAD

**Issue by dwightguth** [kframework/k-legacy#1674]
Currently many definitions that are correct work correctly. The number of incorrect definitions that report a good error message is much lower. We need to come up with an integration test to test a number of the more common error messages in K to ensure that they are caught and given good error messages no matter what.

**Comment by dwightguth** [comment]
Why was this removed from the milestone? I get that it's not part of the tutorial itself, but it's still vitally needed before we can release K 4.0 and get rid of the old pipeline.

Also, one error in particular I want to make sure we catch is that we need to have a good error message if the user accidentally makes a typo and includes a klabel in their definition that is never declared in a production.
